webpack/assets/javascripts/react_app/components/AutoCompleteSearch/AutoComplete.js

```diff
@@ -0,0 +1,108 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import { Icon, OverlayTrigger, Tooltip, TypeAheadSelect } from 'patternfly-react';
+import { STATUS } from '../../constants';
+import { bindMethods, debounceMethods } from '../../common/helpers';
+import './auto-complete.scss';
+
+class AutoComplete extends React.Component {
+  constructor(props) {
+    super(props);
+    bindMethods(this, ['handleClear', 'handleInputChange', 'handleItemSelect', 'handleInputFocus', 'getResults']);
+    debounceMethods(this, 250, ['handleInputChange', 'handleInputFocus', 'handleLoading']);
+  }
+
+  getResults(query, trigger) {
+    const { getResults, data: { controller, url } } = this.props;
+    getResults(url, query, controller, trigger);
+  }
+
+  handleInputFocus({ target: { value } }) {
+    // Only on initial focus.
+    if (this.props.results.length === 0) {
+      this.getResults(value, 'Input focus');
+    }
+  }
+
+  handleInputChange(query) {
+    this.getResults(query, 'Input changed');
+  }
+
+  handleItemSelect(selectedOptions) {
+    this.getResults(selectedOptions[0], 'Item selected');
+    /**
+       *  HACK: I had no choice but to call to an inner function,
+       * due to lack of design in react-bootstrap-typeahead.
+        */
+    this._typeahead.getInstance()._showMenu();
```

Comment:
`showMenu` is not a [public method](https://github.com/ericgio/react-bootstrap-typeahead/blob/master/docs/Methods.md), it might cause issues.